lib/net/NodeHttpClient.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const http = require('http');
+const https = require('https');
+
+const HttpClient = require('./HttpClient');
+
+const defaultHttpAgent = new http.Agent({keepAlive: true});
+const defaultHttpsAgent = new https.Agent({keepAlive: true});
```

Comment:
Is there a reason not to move this into the constructor? It'd be nice to not have side effects on import in the case that this is being imported in a non-node environment.

Reply:
We do need a singleton -- if the user instantiates multiple stripe clients (without overriding the agent) then they should all share the same connection pool. We could lazily instantiate it when the constructor is called, though.

Reply:
I think this is resolved by @dcr-stripe 's latest changes